File: app/src/main/java/com/jdjv/smsbot/BootReceiver.java
package com.jdjv.smsbot;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            if (context.getSharedPreferences("smsbot", Context.MODE_PRIVATE)
                    .getBoolean("enabled", true)) {
                context.startService(new Intent(context, SmsBotService.class));
            }
        }
    }
}

File: app/src/main/java/com/jdjv/smsbot/BootReceiver.kt
package com.jdjv.smsbot

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            val prefs = context.getSharedPreferences("smsbot", Context.MODE_PRIVATE)
            if (prefs.getBoolean("enabled", true)) {
                val svc = Intent(context, SmsBotService::class.java)
                context.startForegroundService(svc)
            }
        }
    }
}
